File: frontend/admin/packages/modules/system/src/api/system/admin/v1/base_table_archive_restore.ts
import service from "@liujitcn/kratos-admin-core/request";
import type { Empty } from "@liujitcn/kratos-admin-system/rpc/google/protobuf/empty";
import type { BaseTableArchiveRestoreService } from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_table_archive_restore";
import type {
  BaseTableArchiveRestore,
  ExecuteBaseTableArchiveRestoreRequest,
  GetBaseTableArchiveRestoreRequest,
  PageBaseTableArchiveRestoreRequest,
  PageBaseTableArchiveRestoreResponse
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_table_archive_restore";

const BASE_TABLE_ARCHIVE_RESTORE_URL = "/v1/admin/base/table-archive-restore";

/** 表归档恢复服务实现。 */
export class BaseTableArchiveRestoreServiceImpl implements BaseTableArchiveRestoreService {
  /** 分页查询归档恢复记录。 */
  PageBaseTableArchiveRestore(request: PageBaseTableArchiveRestoreRequest): Promise<PageBaseTableArchiveRestoreResponse> {
    return service<PageBaseTableArchiveRestoreRequest, PageBaseTableArchiveRestoreResponse>({
      url: BASE_TABLE_ARCHIVE_RESTORE_URL,
      method: "get",
      params: request
    });
  }

  /** 查询归档恢复记录。 */
  GetBaseTableArchiveRestore(request: GetBaseTableArchiveRestoreRequest): Promise<BaseTableArchiveRestore> {
    return service<GetBaseTableArchiveRestoreRequest, BaseTableArchiveRestore>({
      url: `${BASE_TABLE_ARCHIVE_RESTORE_URL}/${request.id}`,
      method: "get"
    });
  }

  /** 手工执行归档恢复。 */
  ExecuteBaseTableArchiveRestore(request: ExecuteBaseTableArchiveRestoreRequest): Promise<Empty> {
    return service<BaseTableArchiveRestore | undefined, Empty>({
      url: BASE_TABLE_ARCHIVE_RESTORE_URL,
      method: "post",
      data: request.base_table_archive_restore
    });
  }
}

/** 表归档恢复服务实例。 */
export const defBaseTableArchiveRestoreService = new BaseTableArchiveRestoreServiceImpl();

File: backend/internal/service/system/admin/v1/base_table_archive_restore.go
package admin

import (
	"context"
	"fmt"

	"github.com/go-kratos/kratos/v3/log"
	adminv1 "github.com/liujitcn/kratos-admin/backend/api/gen/go/system/admin/v1"
	biz "github.com/liujitcn/kratos-admin/backend/internal/biz/system/admin"
	"github.com/liujitcn/kratos-core/errorsx"
	"google.golang.org/protobuf/types/known/emptypb"
)

// BaseTableArchiveRestoreService 手工恢复归档数据。
type BaseTableArchiveRestoreService struct {
	adminv1.UnimplementedBaseTableArchiveRestoreServiceServer
	baseTableArchiveRestoreCase *biz.BaseTableArchiveRestoreCase
}

// NewBaseTableArchiveRestoreService 创建归档恢复服务。
func NewBaseTableArchiveRestoreService(baseTableArchiveRestoreCase *biz.BaseTableArchiveRestoreCase) *BaseTableArchiveRestoreService {
	return &BaseTableArchiveRestoreService{baseTableArchiveRestoreCase: baseTableArchiveRestoreCase}
}

// PageBaseTableArchiveRestore 分页查询归档恢复记录。
func (s *BaseTableArchiveRestoreService) PageBaseTableArchiveRestore(ctx context.Context, req *adminv1.PageBaseTableArchiveRestoreRequest) (*adminv1.PageBaseTableArchiveRestoreResponse, error) {
	result, err := s.baseTableArchiveRestoreCase.PageBaseTableArchiveRestore(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageBaseTableArchiveRestore %v", err))
		return nil, errorsx.WrapInternal(err, "查询归档恢复记录失败")
	}
	return result, nil
}

// GetBaseTableArchiveRestore 查询归档恢复记录。
func (s *BaseTableArchiveRestoreService) GetBaseTableArchiveRestore(ctx context.Context, req *adminv1.GetBaseTableArchiveRestoreRequest) (*adminv1.BaseTableArchiveRestore, error) {
	result, err := s.baseTableArchiveRestoreCase.GetBaseTableArchiveRestore(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("GetBaseTableArchiveRestore %v", err))
		return nil, errorsx.WrapInternal(err, "查询归档恢复记录失败")
	}
	return result, nil
}

// ExecuteBaseTableArchiveRestore 手工创建归档恢复请求。
func (s *BaseTableArchiveRestoreService) ExecuteBaseTableArchiveRestore(ctx context.Context, req *adminv1.ExecuteBaseTableArchiveRestoreRequest) (*emptypb.Empty, error) {
	err := s.baseTableArchiveRestoreCase.ExecuteBaseTableArchiveRestore(ctx, req.GetBaseTableArchiveRestore())
	if err != nil {
		log.Error(fmt.Sprintf("ExecuteBaseTableArchiveRestore %v", err))
		return nil, errorsx.WrapInternal(err, "执行归档恢复失败")
	}
	return new(emptypb.Empty), nil
}
